scipy/stats/_entropy.py

```diff
@@ -196,7 +257,60 @@ def differential_entropy(
     logs = np.log(
         len(differences) * differences / (2 * window_length),
     )
-    if base is not None:
-        logs /= np.log(base)
 
     return np.mean(logs, axis=0)
+
+
+def _pad_along_last_axis(X, m):
+    """Pad the data for computing the rolling window difference"""
+    # scales a  bit better than method in _vasicek_like_entropy
+    shape = np.array(X.shape)
+    shape[-1] = m
+    Xl = np.broadcast_to(X[..., [0]], shape)  # [0] vs 0 to maintain shape
+    Xr = np.broadcast_to(X[..., [-1]], shape)
+    return np.concatenate((Xl, X, Xr), axis=-1)
+
+
+def _van_es_entropy(X, m):
+    """Compute the van Es estimator as described in [6]"""
+    # No equation number, but referred to as HVE_mn.
+    # Typo: there should be a log within the summation.
+    n = X.shape[-1]
+    term1 = 1/(n-m) * np.sum(np.log((n+1)/m * (X[..., m:] - X[..., :-m])),
+                             axis=-1)
+    k = np.arange(m, n+1)
+    return term1 + np.sum(1/k) + np.log(m) - np.log(n+1)
+
+
+def _ebrahimi_entropy(X, m):
+    """Compute the Ebrahimi estimator as described in [6]"""
+    # No equation number, but referred to as HE_mn
+    n = X.shape[-1]
+    X = _pad_along_last_axis(X, m)
+
+    differences = X[..., 2 * m:] - X[..., : -2 * m:]
+
+    i = np.arange(1, n+1).astype(float)
+    ci = np.ones_like(i)*2
+    ci[i <= m] = 1 + (i[i <= m] - 1)/m
+    ci[i >= n - m + 1] = 1 + (n - i[i >= n-m+1])/m
```

Comment:
It may or may not be faster, but yes, it is the most common case, so your understanding is correct. Mostly I did it to eliminate one line from the logic. I would use a function `twos_like` if I could; this is the closest thing to that.

Reply:
Ok thanks for confirming this.